cmd/kubeadm/app/master/selfhosted_etcd.go

```diff
@@ -164,6 +173,69 @@ func CreateSelfHostedEtcdCluster(cfg *kubeadmapi.MasterConfiguration, client *cl
 	return nil
 }
 
+func createTLSAssets(client *clientset.Clientset) error {
+	if err := createTLSSecret(client, operatorSecret, "etcd", []string{}); err != nil {
+		return err
+	}
+
+	peerDNSNames := []string{
+		fmt.Sprintf("*.%s.%s.svc.cluster.local", etcdCluster, metav1.NamespaceSystem),
+	}
+	if err := createTLSSecret(client, peerSecret, "peer", peerDNSNames); err != nil {
+		return err
+	}
+
+	clientDNSNames := []string{
+		fmt.Sprintf("*.%s.%s.svc.cluster.local", etcdCluster, metav1.NamespaceSystem),
+		fmt.Sprintf("*.%s-client.%s.svc.cluster.local", etcdCluster, metav1.NamespaceSystem),
```

Comment:
It should be `%s-client.%s.svc.cluster.local`.
See
https://github.com/coreos/etcd-operator/blob/master/doc/user/cluster_tls.md#memberclientsecret .

Additionally, please beware of that the CA for this key-cert pair should be used by etcd client side. And put the etcd client's CA for peer's client port.